aten/src/ATen/VmapTransforms.cpp

```diff
@@ -94,33 +94,112 @@ static BatchDims computeFrontBatchDimsFromLevels(std::bitset<kVmapNumLevels> lev
   return bdims;
 }
 
-static bool batchDimsAreSameSize(
```

Comment:
The diff view in github doesn't make it clear what happened, so I am explaining it here.

Previously, `VmapPhysicalView::newLogicalFromPhysical(physical)` would check that leading dimensions of `physical` have the same sizes as the batch dimensions of VmapPhysicalView's tensor.

However, I wanted to reuse `VmapPhysicalView` as the output for `BroadcastingVmapTransform`. BroadcastingVmapTransform returns physical views that don't have to have the same batch dimension sizes because we know that the underlying operation is able to broadcast the dimensions anyways. For example, given inputs of size (B0,) and (B0, B1), where B0 and B1 are batch dimensions, `BroadcastingVmapTransform::logicalToPhysical` returns two tensors of size `(B0, 1)` and `(B0, B1)`. Later, the broadcasting operation (like `torch::mul`) can happily multiply two tensors of those size together because it handles the broadcasting under the hood.